src/window.py

```diff
@@ -721,6 +768,46 @@ def TransformActor(obj, event):
             # self.interactor.GetInteractorStyle().SetCurrentStyleToTrackballCamera()
         self.reload_scene()
 
+    def save_current_movement(self):
+        movements = self.stlActor.movements_array
+        current_index = self.stlActor.current_movement_index
+
+        movements = [movement for movement in movements if movement[0] <= current_index]
+
+        movement = self.stlActor.GetUserTransform()
+        movements.append((current_index + 1, movement))
+
+        max_movements = 100
+        if len(movements) > max_movements:
+            movements.pop(0)
+
+        self.stlActor.movements_array = movements
+        self.stlActor.current_movement_index = len(movements) - 1
+
+    def cancel_movement(self, cancel=True):
```

Comment:
It looks like we do not only cancel movements here. Maybe it is shift movement or something else? Then, `cancel` will have a better meaning